make/libraries

```diff
@@ -49,20 +50,24 @@ $(SUNDIALS)/lib/libsundials_nvecserial.a: $(SUNDIALS_NVECSERIAL)
 	@mkdir -p $(dir $@)
 	$(AR) -rs $@ $^
 
-LIBSUNDIALS = $(SUNDIALS)/lib/libsundials_nvecserial.a $(SUNDIALS)/lib/libsundials_cvodes.a $(SUNDIALS)/lib/libsundials_idas.a
+LIBSUNDIALS := $(SUNDIALS)/lib/libsundials_nvecserial.a $(SUNDIALS)/lib/libsundials_cvodes.a $(SUNDIALS)/lib/libsundials_idas.a
 
-STAN_CVODES_HEADERS := $(shell find stan -name *cvodes*.hpp)
-$(STAN_CVODES_HEADERS) : $(LIBSUNDIALS)
+STAN_SUNDIALS_HEADERS := $(shell find stan -name *cvodes*.hpp) $(shell find stan -name *idas*.hpp)
```

Comment:
The sundials things do not work as advertised, I think. For one, you are using the `COMPILE.c`, but I think you meant the `COMPILE.cxx` (otherwise you could drop the `-x c` argument). Moreover, the `CXXFLAGS_SUNDIALS` are not used right now.

Reply:
That's why I asked. I intentionally used `COMPILE.c` because it's a C process.

Questions:
1.  Can you verify that it doesn't work? (Because it does on my computer.) If it doesn't, what configuration do you have?
2. I'll double check the sundials flags. We don't need those flags in the regular build, right? Just for sundials?
3. Should we use `COMPILE.cxx` with the additional `-x c` or should we be using `COMPILE.c` without `-x c`? Is there a difference?

Reply:
It does compile, but all those extra flags we set are being ignored, because we setup the C++ compiler and not the C compiler. So in order:

1. It compiles, but only with the vanilla C compiler. None of our additional flags are set. Thus things will probably fail on windows 32 or 64 bit, because this stuff is setup for the C++ compiler - not the C compiler.

2. Not sure what you mean. Do you refer to CXXFLAGS_SUNDIALS? Those will probably be only needed for the sundials build, yes. Maybe we rename this to CFLAGS_SUNDIALS? It's C after all.

3. The clean way is to use `COMPILE.c` as you did it... but we are carefully configuring the C++ compiler! It is easier for us to use the `COMPILE.cxx` with the `-x c` flag. The super clean way is (I think) to also setup carefully the C compiler (just as the C++ compiler) and then use `COMPILE.c`. The thing is that the `-x c` stuff is supported by `gcc` and `clang` (and I think Intel as well), but I am not aware of that this is commonly agreed on standard. So the `-x c` seems to be a pragmatic solution here instead of setting up C++ and a C compiler.

Reply:
Sorry, my replies weren't threaded. I got it to work, but it doesn't use all the flags. Although `-x c` is supported, the flags for language (e.g. `-std=c++1y`) breaks the compilation. So I had to break down the C++ compiler flags so it's more barebones. I don't know if it's better or worse than just using the C compiler, but I think it's fine and it matches the old behavior.